include/macis/gf/bandlan.hpp

```diff
@@ -0,0 +1,282 @@
+ * @date 25/04/2022
+ */
+bool QRdecomp_tr(std::vector<double> &Q, std::vector<double> &R, int Qrows,
+                 int Qcols);
+
+/**
+ * @brief Wrapper to LAPACK routine to evaluate the eigenvectors
+ *        and eigenvalues of the symmetric matrix mat.
+ *
+ * @param[inout] std::vector<double> &mat: Matrix for
+ *               which to compute the eigenvalues/vectors. Erased
+ *               during computation.
+ * @param[out] std::vector<double> &eigvals: Eigenvalues, sorted from smallest
+ *             to largest.
+ * @param[out] std::vector<double> &eigvecs: Eigenvectors,
+ *             stored as row vectors.
+ * @param[in] int matsize: Nr. of rows/columns of the square matrix mat.
+ *
+ * @returns bool: Error code from LAPACK.
+ *
+ * @author Carlos Mejuto Zaera
+ * @date 25/04/2022
+ */
+bool GetEigsys(std::vector<double> &mat, std::vector<double> &eigvals,
+               std::vector<double> &eigvecs, int matsize);
+
+/**
+ * @brief Wrapper to LAPACK routine to evaluate the eigenvectors
+ *        and eigenvalues of the symmetric band matrix mat.
+ *
+ * @param[inout] std::vector<double> &mat: Matrix for
+ *               which to compute the eigenvalues/vectors. Erased
+ *               during computation.
+ * @param[in] int nSupDiag: Nr. of bands.
+ * @param[out] std::vector<double> &eigvals: Eigenvalues, sorted from smallest
+ *             to largest.
+ * @param[out] std::vector<double> &eigvecs: Eigenvectors,
+ *             stored as row vectors.
+ * @param[in] int matsize: Nr. of rows/cols of the square matrix mat.
+ *
+ * @returns bool: Error code from LAPACK.
+ *
+ * @author Carlos Mejuto Zaera
+ * @date 25/04/2022
+ */
+bool GetEigsysBand(std::vector<double> &mat, int nSupDiag,
+                   std::vector<double> &eigvals, std::vector<double> &eigvecs,
+                   int matsize);
+
+/**
+ * @brief Perform a band Lanczos calculation on the Hamiltonian operator H,
+ * starting from vectors qs, for at most nLanIts iterations. The resulting
+ * band-diagonal matrix Hamiltonian will be stored in bandH. Note that this
+ * implementation does not account for deflations (i.e., pruning the span of the
+ * qs for linear dependencies in higher powers of H).
+ *
+ * @param[in] const sparseexx::csr_matrix<double, int32_t> &H: Hamiltonian
+ * oprator. Just needs to implement a matrix vector product.
+ * @param[in] std::vector<Cont> &qs: Initial set of vetors to
+ * perform the band Lanczos on. Deleted on exit.
+ * @param[in] std::vector<Cont> &bandH: On exit, band-diagonal
+ * Hamiltonian approximation.
+ * @param[inout] int &nLanIts: Number of Lanczos iterations to perform.
+ * @param[in] double thres: Threshold determining when to ignore beta's for
+ * being too small.
+ * @param[in] bool print: If true, write intermediate results to file.
+ *
+ * @author Carlos Mejuto Zaera
+ * @date 25/04/2022
+ */
+template <class Cont, class Functor>
+void MyBandLan(const Functor &H, std::vector<Cont> &qs,
```

Comment:
Naming conventions of `My<Func>` should get swapped out for something more meaningful. I think `BandLan` works fine here.